spec/mailers/user_mailer_spec.rb

```diff
@@ -311,4 +311,52 @@
       end
     end
   end
+
+  describe "invite request declined" do
+    before(:each) do
+      @user = FactoryGirl.create(:user)
+      @total = 2
+      @reason = "You smell"
+    end
+
+    let(:email) { UserMailer.invite_request_declined(@user.id, @total, @reason).deliver }
+
+    # Test the headers
+    it 'has a valid from line' do
+      text = "Archive of Our Own <#{ArchiveConfig.RETURN_ADDRESS}>"
+      expect(email.header['From'].to_s).to eq(text)
+    end
+
+    it 'has the correct subject line' do
+      text = "[#{ArchiveConfig.APP_SHORT_NAME}] Additional Invite Code Request Declined"
+      expect(email.subject).to eq(text)
+    end
+
+    # Test both body contents
+    it_behaves_like "multipart email"
+
+    describe 'HTML version' do
+      it 'has text contents' do
+        expect(get_message_part(email, /html/)).to include("We regret to inform you")
```

Comment:
It looks like the tests are set up to simulate the user requesting two invitations. Instead of checking for the text "We regret to inform you," which is the same no matter how many emails are sent, it might be a good idea to check for "new invitations cannot be fulfilled" instead, just to make sure we're getting the correct text for this scenario.

Reply:
Yes, good point.